src/System.Private.CoreLib/src/System/Threading/WaitHandle.cs

```diff
@@ -147,19 +147,21 @@ private bool WaitOneCore(int millisecondsTimeout)
         private static SafeWaitHandle[] ObtainSafeWaitHandles(
             RuntimeThread currentThread,
             WaitHandle[] waitHandles,
+            int numWaitHandles,
```

Comment:
These asserts have been updated. The only one that uses `waitHandles.Length` is one I added in this PR to make sure that the value for `numWaitHandles` won't cause us to go out of bounds.